sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -450,6 +450,7 @@ object GpuOverrides {
       case DateType => true
       case TimestampType => ZoneId.systemDefault().normalized() == GpuOverrides.UTC_TIMEZONE_ID
       case StringType => true
+      case dt: DecimalType if dt.precision <= ai.rapids.cudf.DType.DECIMAL64_MAX_PRECISION => true
```

Comment:
Do we have a list of all of the operators we currently support that also support Decimal in Spark? We have to be sure that if we are allowing decimal for all operators, like this is doing, that we have disallowed it everywhere that we do not have automated tests for.

The only tests I see in this patch are unit tests for Scalars.

Reply:
We will be able to support binary operators for decimal type only if we enable `BinaryOperable` with decimal  on cuDF java layer.
In terms of unary operators (abs, ceil, floor, round, …) and other advanced operators(such as: aggregation operators), we have to wait the support in libcudf level. This refers to https://github.com/rapidsai/cudf/issues/3556.

Reply:
I've rewrited `isSupportedType` part. In latest version, general method `isSupportedType` remain unchanged. And a decimal specialized method `isSupportedTypeWithDecimal` was added for GpuExpressions (operators) who support Deimcal. For now, only very limited ones are decimal-enabled, such as: GpuLiteral, GpuAlias. But this collection will be expanded only if  decimal-related dependencies are ready.

Reply:
I found it is not easy to add DecimalType into supported type while keeping compatible with existed RapidsMetas. I tried to solve this problem with two actions:
1. Add method `isSupportedTypeWithDecimal` for RapidMeta who overrides the `areAllSupportedTypes` interface, such as: `Literal`, `Cast`.
2. Refactor `GpuOverrides.areAllSupportedTypes`, which receives RapidsMeta itself to determine whether it is decimal supportable. This is for RapidsMeta without  interface overriding of `areAllSupportedTypes`.